mule-extensions-api-persistence/src/main/java/org/mule/runtime/extension/api/persistence/metadata/ComponentResultJsonSerializer.java

```diff
@@ -255,4 +243,53 @@ TypeMetadata getAttributes() {
       return success(descriptor);
     }
   }
+
+
+  /**
+   * DTO that represents a {@link OutputMetadataDescriptor} into a serializable format.
+   *
+   * @since 1.0
+   */
+  private static class InputMetadata implements Descriptable<InputMetadataDescriptor> {
+
+    private final List<ParameterMetadata> parameters;
+
+    InputMetadata(ComponentMetadataDescriptor result) {
+      InputMetadataDescriptor intputDescriptor = result.getInputMetadata().get();
+
+      parameters = ImmutableList.<ParameterMetadata>builder()
```

Comment:
why don't you just do a copyOf the collected list? This seems overly complicated